cvss3/base_metrics.go

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cvss3
+
+import "fmt"
+
+type BaseMetrics struct {
+	AttackVector
+	AttackComplexity
+	PrivilegesRequired
+	UserInteraction
+	Scope
+	Confidentiality
+	Integrity
+	Availability
+}
+
+type AttackVector int
+
+const (
+	AttackVectorNetwork AttackVector = iota + 1
+	AttackVectorAdjecent
+	AttackVectorLocal
+	AttackVectorPhysical
+)
+
+var (
+	weightsAttackVector = []float64{0, 0.85, 0.62, 0.55, 0.2}
+	stringAttackVector  = []string{"", "N", "A", "L", "P"}
+	mapAttackVector     = map[string]AttackVector{
+		"N": AttackVectorNetwork,
+		"A": AttackVectorAdjecent,
+		"L": AttackVectorLocal,
+		"P": AttackVectorPhysical,
+	}
+)
+
+func (av AttackVector) Defined() bool {
```

Comment:
Is it supposed to be used outside `package cvss3`?

Reply:
no